app/src/auto_validator/core/utils/utils.py

```diff
@@ -53,3 +61,235 @@ def get_user_ip(request):
+    stdin, stdout, stderr = ssh.exec_command(command)
+    print(stdout.read().decode())
+    print(stderr.read().decode())
+
+
+def generate_pre_config_file(subnet_codename: str, blockchain, netuid, yaml_file_path: str, csv_file_path: str):
+    """
+    Generate pre_config.json file from the YAML file and CSV file.
+
+    Args:
+        subnet_codename (str): Subnet codename
+        yaml_file_path (str): Path to the YAML file
+        csv_file_path (str): Path to the CSV file
+
+    Examples:
+        generate_pre_config_file("subnet1", "mainnet", "1", "subnets.yaml", "secrets.csv")
+    """
+    yaml_file_path = os.path.abspath(yaml_file_path)
+    csv_file_path = os.path.abspath(csv_file_path)
+    pre_config_path = os.path.abspath(f"{LOCAL_SUBNETS_SCRIPTS_PATH}/{subnet_codename}/pre_config.json")
+    with open(yaml_file_path) as file:
+        data = yaml.safe_load(file)
+    if subnet_codename not in data:
+        raise ValueError(f"Subnet codename {subnet_codename} not found in YAML file.")
+    allowed_secrets = data[subnet_codename].get("allowed_secrets", [])
+
+    secrets = {}
+    with open(csv_file_path) as csv_file:
+        reader = csv.DictReader(csv_file)
+        for row in reader:
+            if row["SECRET_KEYS"] in allowed_secrets:
+                if row["SECRET_VALUES"] == "random":
+                    secrets[row["SECRET_KEYS"]] = os.urandom(32).hex()
+                else:
+                    secrets[row["SECRET_KEYS"]] = row["SECRET_VALUES"]
+    secrets["SUBNET_CODENAME"] = subnet_codename
+    secrets["BITTENSOR_NETWORK"] = "finney" if blockchain == "mainnet" else "test"
+    secrets["BITTENSOR_NETUID"] = netuid
+
+    with open(pre_config_path, "w") as file:
+        json.dump(secrets, file, indent=4)
+
+    return pre_config_path
+
+
+def install_validator_on_remote_server(
+    subnet_codename,
+    blockchain,
+    netuid,
+    coldkey_path,
+    hotkey_name,
+    ssh_ip_address,
+    ssh_user,
+    ssh_key_path,
+    ssh_passphrase,
+):
+    """
+    Install a validator on a remote server.
+
+    Args:
+        subnet_codename (str): Subnet codename.
+        blockchain (str): Blockchain name.
+        netuid (str): NetUID.
+        coldkey_path (str): Coldkey path.
+        hotkey_name (str): Hotkey name.
+        ssh_ip_address (str): SSH IP address.
+        ssh_user (str): SSH username.
+        ssh_key_path (str): SSH key path.
+        ssh_passphrase (str): SSH passphrase.
+
+    Examples:
+        install_validator_on_remote_server(
+            "subnet1", "mainnet", "1", "/root/.bittensor/wallets/validator/", "validator-hotkey", "29.92.12.1", "root", "/root/.ssh/id_rsa", "1234567890"
+        )
+    """
+    ssh = create_ssh_client(ssh_ip_address, ssh_user, ssh_key_path, ssh_passphrase)
+
+    yaml_file_path = f"{LOCAL_SUBNETS_SCRIPTS_PATH}/subnets.yaml"
```

Comment:
```suggestion
    yaml_file_path = LOCAL_SUBNETS_SCRIPTS_PATH / "subnets.yaml"
```
after migrating `LOCAL_SUBNETS_SCRIPTS_PATH` to pathlib